man/fi_peer.3.md

```diff
@@ -494,9 +502,17 @@ These calls are invoked by the peer provider to obtain the receive buffer(s)
 where an incoming message should be placed.  The peer provider will pass in
 the relevant fields to request a matching rx_entry from the owner.  If source
 addressing is required, the addr will be passed in; otherwise, the address will
-be set to FI_ADDR_NOT_AVAIL. The size parameter is needed by the owner for
-adjusting FI_MULTI_RECV entries. The peer provider is responsible for checking
-that an incoming message fits within the provided buffer space.
+be set to FI_ADDR_NOT_AVAIL.
+The msg_size field indicates the received message size.
+This field may be needed by the owner when handling FI_MULTI_RECV or FI_PEEK.
+The owner will set the peer_entry->msg_size field on get_msg/tag() for the owner
+and peer to use later, if needed. This field will be set on both the expected
+and unexpected paths.
+The returned rx_entry->iov returned from the owner refers to the full size of
+the posted receive passed to the peer. The peer provider is responsible for
+checking that an incoming message fits within the provided buffer space and
+generating truncation errors.
+The tag parameter is used for tagged messages.
```

Comment:
I would consider documenting that tag must be set to 0 for non-tagged messages, so it's not left uninitialized.

Reply:
Good call. Will update.